src/compiler/binder.ts

```diff
@@ -2503,8 +2503,27 @@ namespace ts {
             }
         }
 
+        function needParameterStrictModeCheck(node: Node) {
+            if (node.parent.kind === SyntaxKind.ConstructorType || node.parent.kind === SyntaxKind.FunctionType) {

```

Comment:
i do not think this is important. i would say if this is your code, and this is a .ts file and your writing an interface with `arguments`, change it.

the issue is more of a problem with .d.ts files, so i would just use `isInAmbientContext`.